### ISSUE TITLE
Dns settings in network stanza not usefull due incompatibility with docker task driver

### ISSUE THREAD

**Issue by tantra35** [hashicorp/nomad#8757]
### Nomad version
Nomad v0.12.3 (e6e969ad7e5c83d44758c982cfb756e081f779c3+CHANGES)

### Issue
when dns settings using with bridge in in network stanza allocs can't start with docker driver

```
Recent Events:
Time                       Type            Description
2020-08-27T17:29:10+03:00  Killing         Sent interrupt. Waiting 1m0s before force killing
2020-08-27T17:29:10+03:00  Not Restarting  Error was unrecoverable
2020-08-27T17:29:09+03:00  Driver Failure  failed to create container: API error (400): conflicting options: dns and the network mode
2020-08-27T17:29:07+03:00  Driver          Downloading image
2020-08-27T17:29:07+03:00  Task Setup      Building Task Directory
2020-08-27T17:29:05+03:00  Received        Task received by client
```

### Job file (if appropriate)
```
job "vault_debug-00"
{
        datacenters = ["test"]
        priority = 50

        update
        {
                stagger = "30m"
                max_parallel = 1
        }

        group test
        {
                count = 1

                network
                {
                        mode = "bridge"
                        dns {
                          servers = ["172.17.0.1"]
                        }
                }

                task "vault_debug_task_docker"
                {
                        driver = "docker"
                        kill_timeout = "1m"

                        config
                        {
                                image = "redis"
                        }

                        resources
                        {
                                memory = 200
                                cpu = 200
                        }
                }

                task "vault_debug_task_exec"
                {
                        driver = "exec"
                        kill_timeout = "1m"

                        config
                        {
                                command = "sleep"
                                args = ["600"]
                        }

                        resources
                        {
                                memory = 200
                                cpu = 200
                        }
                }
        }
}
```

**Comment by shoenig** [comment]
Sorry about that @tantra35 ! We should already have a fix for this in https://github.com/hashicorp/nomad/pull/8600 going out in the next release.

**Comment by tgross** [comment]
This issue was closed by #8600, released in 0.12.4. (Changelog entry was missing but I've added it in https://github.com/hashicorp/nomad/pull/9665)
